### ISSUE TITLE
delete children nodes

### ISSUE THREAD

**Issue by sinisarudan** [mprinc/KnAllEdge#161]
- when ever we delete a node, we should; delete all of his **descendant-nodes** (and accompanied **edges**) **that don't have other parent** (i.e, that would anyway become inaccessible).
- **Should we ask/warn** in this cases of multiple deleting ?

Explanation - which nodes to preserve from deleting:
![screen shot 2016-07-12 at 12 32 53 pm](https://cloud.githubusercontent.com/assets/1769627/16764166/e8cf0248-482c-11e6-841d-d88c4580136e.png)

**Comment by mprinc** [comment]
Yes we should ask, or at least offer undo :)

**Comment by mprinc** [comment]
Yes we should ask, or at least offer undo :)

**Comment by sinisarudan** [comment]
yes, it's just slow down a bit ths procedure, but we can pre-select button "OK" on confirmation dialog :)
